### ISSUE TITLE
Dynamically Force Opening A Circuit on a javanica @HystrixCommand request 

### ISSUE THREAD

**Issue by Npscholar** [Netflix/Hystrix#1627]
This is primarily a question and not necessarily an issue. 

When using the @HystrixCommand and configuring HystrixCommandProperties we are able to set 

```
@HystrixCommand(commandKey = "blah",
 fallbackMethod = "fallblah",
      commandProperties = {@HystrixProperty(name = "circuitBreaker.forceOpen", value="true")},
            threadPoolProperties = {@HystrixProperty(name = "coreSize", value = "100"),
          @HystrixProperty(name = "maxQueueSize", value = "101")})
```

However since HystrixCommandProperties is a Runtime annotation it only accepts constants as parameters forcing me to lose any dynamic property configuration benefits (e.g. force opening a circuit w/o a redeploy or restart).

1. Is this a known limitation ?
2. Was it by design ? 
3. Is there any other way to dynamically configure the hystrix config properties that javanica uses to create a HystrixCommand ?

Any suggestion you could provide would be greatly appreciated 
Thanks

**Comment by mattrjacobs** [comment]
Runtime configuration is achieved within Hystrix by using Archaius: https://github.com/Netflix/Hystrix/wiki/Configuration#intro

Does that answer your question?

**Comment by Serotonin7** [comment]
Hey @mattrjacobs  thanks for your response. I understand how archaius and hystrix work together when not using javanica. 

My question is around how do I get that same runtime configuration while using javanica? I am not seeing anywhere in the javanica code that it pulls from a property file location during runtime. It just takes the properties defined in the annotation argument and overrides (or not) the initial prop value.

With that approach a code change and restart is required to change the hystrix config

cc @dmgcodevil

**Comment by dmgcodevil** [comment]
Yes so far as I know the only way to set properties is to use annotation ,
not aware of other options in hystrix. Will investigate and implement
similar approach in javanica if it's possible .

**Comment by Serotonin7** [comment]
Thanks @dmgcodevil. I am interesting in leveraging javanica in some future projects however I cannot afford to lose the ability to configure runtime circuits without a code change and redeploy. If you do find a solution for this your effort would be greatly appreciated!

**Comment by rmozone** [comment]
This is sad. It would be great if Hystrix Javanica supported the dynamic configuration via Archaius. When I try to do this, I see a warning saying _"Add a configuration to an existing non-aggregated installed configuration will have no effect"_ and as expected my custom configuration is not loaded.

**Comment by flycun** [comment]
ConfigurationManager
			.getConfigInstance()
			.setProperty("hystrix.command.YourCommandKey.circuitBreaker.forceOpen",	true);
